sap_hana/datadog_checks/sap_hana/sap_hana.py

```diff
@@ -0,0 +1,572 @@
+
+            files_open = service['files_open']
+            self.gauge('file.service.open', files_open, tags=tags, hostname=host)
+
+            cpu_percent = total_time_to_temporal_percent(service['cpu_time'])
+            self.rate('cpu.service.utilized', cpu_percent, tags=tags, hostname=host)
+
+    def query_volume_io(self):
+        # https://help.sap.com/viewer/4fe29514fd584807ac9f2a04f6754767/2.0.02/en-US/20cadec8751910148bab98528e3634a9.html
+        for volume in self.iter_rows(queries.GlobalSystemVolumeIO):
+            tags = [
+                'db:{}'.format(volume['db_name']),
+                'hana_port:{}'.format(volume['port']),
+                'resource_type:{}'.format(volume['resource']),
+                'fs_path:{}'.format(volume['path']),
+            ]
+            tags.extend(self._tags)
+
+            hana_host = volume['host']
+            tags.append('hana_host:{}'.format(hana_host))
+            host = self.get_hana_hostname(hana_host)
+
+            # Read
+            reads = volume['reads']
+            self.gauge('volume.io.read.total', reads, tags=tags, hostname=host)
+            self.monotonic_count('volume.io.read.count', reads, tags=tags, hostname=host)
+
+            read_size = volume['read_size']
+            self.gauge('volume.io.read.size.total', read_size, tags=tags, hostname=host)
+            self.monotonic_count('volume.io.read.size.count', read_size, tags=tags, hostname=host)
+
+            read_percent = total_time_to_temporal_percent(volume['read_time'], scale=MICROSECOND)
+            self.rate('volume.io.read.utilized', read_percent, tags=tags, hostname=host)
+
+            # Write
+            writes = volume['writes']
+            self.gauge('volume.io.write.total', writes, tags=tags, hostname=host)
+            self.monotonic_count('volume.io.write.count', writes, tags=tags, hostname=host)
+
+            write_size = volume['write_size']
+            self.gauge('volume.io.write.size.total', write_size, tags=tags, hostname=host)
+            self.monotonic_count('volume.io.write.size.count', write_size, tags=tags, hostname=host)
+
+            write_percent = total_time_to_temporal_percent(volume['write_time'], scale=MICROSECOND)
+            self.rate('volume.io.write.utilized', write_percent, tags=tags, hostname=host)
+
+            # Overall
+
+            # Convert microseconds -> seconds
+            io_time = volume['io_time'] / MICROSECOND
+            io_percent = total_time_to_temporal_percent(io_time, scale=1)
+            self.rate('volume.io.utilized', io_percent, tags=tags, hostname=host)
+
+            if io_time:
+                throughput = (read_size + write_size) / io_time
+            else:
+                throughput = 0
+            self.gauge('volume.io.throughput', throughput, tags=tags, hostname=host)
+
+    def query_custom(self):
+        for custom_query in self._custom_queries:
+            query = custom_query.get('query')
+            if not query:  # no cov
```

Comment:
Still need `# no cov` ? Tests seems to assert the errors.

Same for other cases.